lib/livebook/hubs.ex

```diff
@@ -54,18 +70,29 @@ defmodule Livebook.Hubs do
   def save_hub(struct) do
     attributes = struct |> Map.from_struct() |> Map.to_list()
     :ok = Storage.insert(@namespace, struct.id, attributes)
+    :ok = connect_hub(struct)
     :ok = broadcast_hubs_change()
+
     struct
   end
 
   @doc false
   def delete_hub(id) do
-    Storage.delete(@namespace, id)
+    with {:ok, hub} <- get_hub(id) do
+      if connected_hub = get_connected_hub(hub) do
+        Process.exit(connected_hub.pid, :normal)
```

Comment:
This is better:

```suggestion
        GenServer.stop(connected_hub.pid, :shutdown)
```

It will block until it stops and `:shutdown` guarantees links will be broken. :) It is fine to assume GenSever for now.